packages/ts/generator-typescript-plugin-model/src/MetadataParser.ts

```diff
@@ -0,0 +1,57 @@
+import type { Schema } from '@hilla/generator-typescript-core/Schema.js';
+import ts, { type ObjectLiteralExpression, type PropertyAssignment } from 'typescript';
+
+interface Annotation {
+  name: string;
+  attributes?: Record<string, unknown>;
+}
+
+export type SchemaWithMetadata = Schema & {
+  'x-annotations'?: Annotation[];
+  'x-java-type'?: string;
+};
+
+export class MetadataParser {
+  parse(schema: Schema): ObjectLiteralExpression | null {
+    const schemaWithMetadata = schema as SchemaWithMetadata;
+
+    const properties = [
+      this.#createAnnotationsProperty(schemaWithMetadata),
+      this.#createJavaTypeProperty(schemaWithMetadata),
+    ].filter(Boolean) as PropertyAssignment[];
+
+    if (properties.length === 0) {
+      return null;
+    }
+
+    return ts.factory.createObjectLiteralExpression(properties);
+  }
+
+  #createAnnotationsProperty(schema: SchemaWithMetadata): PropertyAssignment | null {
+    const annotations = schema['x-annotations'];
+    const hasAnnotations = annotations && annotations.length > 0;
+    if (!hasAnnotations) {
+      return null;
+    }
+
+    const annotationLiterals = annotations.map((annotation) =>
+      ts.factory.createObjectLiteralExpression([
+        ts.factory.createPropertyAssignment('name', ts.factory.createStringLiteral(annotation.name)),
+      ]),
+    );
```

Comment:
Annotations can also have a generic object of attributes, but it wasn't clear how I could get those generated. There is something similar in the validation constraints parser but I wasn't able to adapt that so far. I would suggest to start without the attributes for now, as we don't have a use-case for them at the moment.

Reply:
Aren't they gonna be put in the same object? Like:
```ts
{
  name: "jakarta.persistense.Id",
  param1: 0,
  param2: "string",
}
```

Reply:
I usually use this [site](https://ts-ast-viewer.com) to understand what calls to use if I want to create a specific AST. 

For the object literal, the code should look similar to that one:

```ts
ts.factory.createPropertyAssignment(
  ts.factory.createIdentifier("attributes"),
  ts.factory.createObjectLiteralExpression(
    Object.entries(attributes).map(([name, value]) => ts.factory.createPropertyAssignment(
      ts.factory.createIdentifier(name),
      // ...create specific literal for value, e.g.:
      ts.factory.createStringLiteral(value)
    )),
    true
  )
)
```